### DISCUSSION TITLE
Unable to increase open file limit

### DISCUSSION THREAD

**Discussion by El-Tommy** [rabbitmq/rabbitmq-server#9303] [Other] [answered]
### Describe the bug

OS: Ubuntu 20.04
RabbitMQ: 3.12.4

I'm trying to do everything the documentation suggests to increase RabbitMQ's `open file descriptor`, but it has no effect.

I have already made all the changes below and the result of `rabbitmqctl status` does not change.

```
sudo nano /etc/sysctl.conf
fs.file-max=1048576

sudo nano /etc/default/rabbitmq-server
ulimit -S -n 1048576

sudo nano /etc/rabbitmq/rabbitmq-env.conf
ulimit -S -n 1048576
#This action causes the server not to start
```

This directory and file does not exist: `/etc/systemd/system/rabbitmq-server.service.d/limits.conf`

And the result of `rabbitmqctl status` always looks like it always does:

```
File Descriptors

Total: 6, limit: 32671
Sockets: 0, limit: 29401
```

And my ulimit is like this:
```
$ulimit -Sn
1048576

$ulimit -Hn
1048576
```

### Reproduction steps

1. Edit /etc/sysctl.conf, /etc/default/rabbitmq-server and /etc/rabbitmq/rabbitmq-env.conf
2. Run rabbitmqctl status

### Expected behavior

Show:

```
File Descriptors

Total: 6, limit: 1048576
```

**Comment by michaelklishin** [comment]
> This directory and file does not exist

You need to create it. On all modern distributions such limits are managed via `systemd`.

Before the limit can take effect, you must restart the node (the OS process)

**Comment by lukebakken** [comment, marked as answer]
@El-Tommy use `systemctl edit rabbitmq-server.service` to create an override. You can then add or change `LimitNOFILE`
